ktor-server-rate-limiting/src/commonMain/kotlin/io/github/flaxoos/ktor/server/plugins/ratelimiter/implementations/SlidingWindow.kt

```diff
@@ -0,0 +1,130 @@
+package io.github.flaxoos.ktor.server.plugins.ratelimiter.implementations
+
+import io.github.flaxoos.ktor.server.plugins.ratelimiter.CallVolumeUnit
+import io.github.flaxoos.ktor.server.plugins.ratelimiter.RateLimiter
+import io.github.flaxoos.ktor.server.plugins.ratelimiter.RateLimiterResponse
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.ktor.server.application.ApplicationCall
+import kotlinx.atomicfu.locks.reentrantLock
+import kotlinx.atomicfu.locks.withLock
+import kotlinx.datetime.Clock.System.now
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+
+private val logger = KotlinLogging.logger {}
```

Comment:
I would suggest letting users provide their own logger that adheres to an interface

Reply:
The logger is juts a facade. The underlying logger will be chosen by the user based on the classpath: https://www.baeldung.com/kotlin/kotlin-logging-library